certbot/main.py

```diff
@@ -155,34 +155,39 @@ def _handle_subset_cert_request(config, domains, cert):
             "reinvoke the client.")
 
 
-def _handle_identical_cert_request(config, cert):
-    """Figure out what to do if a cert has the same names as a previously obtained one
+def _handle_identical_cert_request(config, lineage):
+    """Figure out what to do if a lineage has the same names as a previously obtained one
 
-    :param storage.RenewableCert cert:
+    :param storage.RenewableCert lineage:
 
     :returns: Tuple of (str action, cert_or_None) as per _treat_as_renewal
               action can be: "newcert" | "renew" | "reinstall"
     :rtype: tuple
 
     """
-    if renewal.should_renew(config, cert):
-        return "renew", cert
+    if lineage.has_pending_deployment():
```

Comment:
I'm not sure that it's ideal that there's absolutely no way to override this behavior with a command-line flag, although I appreciate the argument that typically users would not want to do so. While this is a good default even in respect to our UI, maybe we should consider having a way to be more explicit about this (although that risks a cascade of `--no-really` flags).